File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        int total = 0;
        for (int x : nums) total += x;
        if (total < target) return 0;

        int l = 0;
        int min_len = INT_MAX;
        int summation = 0;

        for (int r = 0; r < nums.size(); r++) {
            summation += nums[r];

            while (summation > target) {
                if (summation >= target) {
                    min_len = min(min_len, r - l + 1);
                }
                summation -= nums[l];
                l++;
            }

            if (summation >= target) {
                min_len = min(min_len, r - l + 1);
            }
        }

        return min_len;
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        if (sum(nums)<target):
            return 0
        l = 0
        min_len = float("inf")
        summation = 0
        for r in range(len(nums)):
            summation+=nums[r]
            while(summation>target):
                    
                if summation>=target:
                    min_len = min(r-l+1,min_len)
                summation-=nums[l]
                l+=1  
                
            if summation>=target:
                min_len = min(r-l+1,min_len)
  
        return min_len

        